BlazarService/src/main/java/com/hubspot/blazar/resources/RepositoryBuildResource.java

```diff
@@ -41,20 +42,23 @@ public RepositoryBuildResource(RepositoryBuildService repositoryBuildService,
   @POST
   @Path("/branch/{id}")
   public RepositoryBuild trigger(@PathParam("id") int branchId) {
-    return triggerWithOptions(branchId, BuildOptions.defaultOptions());
+    return triggerWithOptions(branchId, BuildOptions.defaultOptions(), Optional.<String>absent());
   }
 
   @POST
   @Path("/branch/{id}")
   @Consumes(MediaType.APPLICATION_JSON)
-  public RepositoryBuild triggerWithOptions(@PathParam("id") int branchId, BuildOptions buildOptions) {
+  public RepositoryBuild triggerWithOptions(@PathParam("id") int branchId, BuildOptions buildOptions, @QueryParam("username") Optional<String> username) {
     Optional<GitInfo> gitInfo = branchService.get(branchId);
     if (!gitInfo.isPresent()) {
       throw new NotFoundException("No branch found with id: " + branchId);
     }
-
-    // TODO capture user
-    BuildTrigger buildTrigger = BuildTrigger.forUser("unknown");
+    BuildTrigger buildTrigger;
+    if (username.isPresent()) {
+      buildTrigger = BuildTrigger.forUser(username.get());
+    }  else {
+      buildTrigger = BuildTrigger.forUser("unknown");
+    }
```

Comment:
You can do `BuildTrigger buildTrigger = BuildTrigger.forUser(username.or("unknown"));`